legion/engine/llri-vk/detail/instance.cpp

```diff
@@ -141,13 +141,16 @@ namespace llri
                 // so instead the check is implicit, implementation callbacks aren't guaranteed
                 if (availableExtensions.find(internal::nameHash("VK_EXT_debug_utils")) != availableExtensions.end())
                 {
-                    extensions.insert(VK_EXT_DEBUG_UTILS_EXTENSION_NAME);
+                    extensions.emplace(internal::nameHash(VK_EXT_DEBUG_UTILS_EXTENSION_NAME), VK_EXT_DEBUG_UTILS_EXTENSION_NAME);
                     output->m_shouldConstructValidationCallbackMessenger = true;
                 }
             }
 
-            std::vector<const char*> extensionVec(extensions.size());
-            std::copy(extensions.begin(), extensions.end(), extensionVec.begin());
+            // transfer over to vector of strings
+            std::vector<const char*> extensionVec;
+            extensionVec.reserve(extensions.size());
+            for (auto& pair : extensions)
+                extensionVec.push_back(pair.second);
```

Comment:
```cpp
std::vector<const char*> extensionVec{extensions.size()};
std::transform(extensions.begin(), extensions.end(), extensionVec.begin(), [](auto& pair){ return pair.second;});
```
how about this? probably won't give any measurable speedup, but theoretically does slightly less? mostly just semantics tho

Reply:
yeah sure, that's a bit cleaner